### PR TITLE
feat(*): show stage log

### PR DESCRIPTION
**Pull request by qiuxiaolian**
**What this PR does / why we need it**:

- add log component
- show stage total log
Screen Shot 2019-12-22 at 15 30 33

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to !-- 填在 Fixes，PR 合并就会关 issue。填在 Reference to 会关联 issue，不会联动关闭。-->

**Special notes for your reviewer**:

/cc @ladjzero @caosm 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: web/src/components/workflowRun/component/detail/index.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+              formatWorkflowRunStage(_.get(detail, 'status.stages')),
+              _.get(detail, 'metadata.annotations.stagePosition')
+            )}
+            project={_.get(params, 'project')}
+            workflowName={_.get(params, 'workflowName')}
+            handleStageLog={this.showStageLog}
+          />
+        </DetailContent>
+        <Modal
+          title={`${activeStage} 的日志`}
```
Comment (ladjzero) [inline web/src/components/workflowRun/component/detail/index.js]: cyclone 国际化怎么处理的？

File: web/src/components/public/log/Log.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    } else {
+      this.logs = this.logs.concat(data);
+    }
+    this.updateThrottle();
+  };
+
+  updateData = () => {
+    const cacheLogs = _.map(this.logs, log => {
+      return {
+        key: this.index++,
```
[the remark this answers is not in the document]
Comment (qiuxiaolian) [reply web/src/components/public/log/Log.js]: 是的，log 没有 ID， log 是一个 string

File: web/src/components/public/log/Log.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+          </div>
+        )}
+      />
+    );
+  };
+
+  render() {
+    const { data } = this.state;
+    return (
+      <div ref={div => (this.logCoreInstance = div)}>
```
[the remark this answers is not in the document]
Comment (qiuxiaolian) [reply web/src/components/public/log/Log.js]: 先去掉，这个暂时没有用到

**Comment by zhujian7** [comment]
The bot hangs up, I will merge the PR manually.

**Pull request merged**